src/main/java/com/amazon/opendistroforelasticsearch/sql/request/SqlRequestFactory.java

```diff
@@ -59,16 +61,36 @@ private static SqlRequest parseSqlRequestFromPayload(RestRequest restRequest) {
         JSONObject jsonContent;
         try {
             jsonContent = new JSONObject(content);
+            if (jsonContent.has(SQL_CURSOR_FIELD_NAME)) {
+                return new SqlRequest(jsonContent.getString(SQL_CURSOR_FIELD_NAME));
+            }
         } catch (JSONException e) {
             throw new IllegalArgumentException("Failed to parse request payload", e);
         }
         String sql = jsonContent.getString(SQL_FIELD_NAME);
+
         if (jsonContent.has(PARAM_FIELD_NAME)) { // is a PreparedStatement
             JSONArray paramArray = jsonContent.getJSONArray(PARAM_FIELD_NAME);
             List<PreparedStatementRequest.PreparedStatementParameter> parameters = parseParameters(paramArray);
             return new PreparedStatementRequest(sql, jsonContent, parameters);
         }
-        return new SqlRequest(sql, jsonContent);
+        return new SqlRequest(sql, validateAndGetFetchSize(jsonContent), jsonContent);
+    }
+
+
+    private static Integer validateAndGetFetchSize(JSONObject jsonContent) {
+        Integer fetchSize = null;
+        try {
+            if (jsonContent.has(SQL_FETCH_FIELD_NAME)) {
+                fetchSize = jsonContent.getInt(SQL_FETCH_FIELD_NAME);
+                if (fetchSize < 0) {
```

Comment:
if fetchSize is initialized to null, will this throw a NullPointerException?

Reply:
This will never throw `NullPointerException`, as we are either assigning integer value to `fetch_size` or throwing `JSONException`